src/dataset/create_dataset.py

```diff
@@ -19,240 +20,141 @@
-            article (Dict[str, Any]): a dictionary containing the article's text and metadata.
-
-        Returns:
-            Tuple[str]: a tuple containing the title, author, date, URL, tags, and text of the article.
-        """
-        title: str = ""
-        author: str = ""
-        date_published: str = None
-        url: str = None
-        tags: str = None
-        text: str = None
-        
-        # Get title
-        if 'title' in article and 'book_title' in article and article['title']: title = article['title']
-        elif 'book_title' in article and 'title' not in article and article['book_title']: 
-            title = article['book_title']
-        elif 'title' in article and article['title']: 
-            title = article['title']
-        title = title.strip('\n').replace('\n', ' ')[:100]
-
-        # Get author
-        if 'author' in article and 'authors' in article and article['author']: author = article['author']
-        elif 'authors' in article and article['authors']: author = article['authors']
-        elif 'author' in article and article['author']: author = article['author']
-        if type(author) == str: author = get_authors_list(author)
-        if type(author) == list: author = ', '.join(author)
-        author = author.strip('\n').replace('\n', ' ')[:100]
-
-        # Get date published
-        if 'date_published' in article and article['date_published'] and len(article['date_published']) >= 10: date_published = article['date_published'][:10]
-        elif 'published' in article and article['published'] and len(article['published']) >= 16: date_published = article['published'][:16]
-        else: date_published = None
-        if date_published is not None:
-            date_published = standardize_date(date_published)
-            
-        # Get URL
-        if 'link' in article and article['link']: url = article['link']
-        elif 'url' in article and article['url']: url = article['url']
-        elif 'doi' in article and article['doi']: url = article['doi']
-        else: url = None
-            
-        # Get tags
-        if 'tags' in article and article['tags']:
-            if type(article['tags']) == list: tags = ', '.join([val['term'] for val in article['tags']])
-            elif type(article['tags']) == str: tags = article['tags']
-            else: tags = None
+                self.entries_per_source_count[source] = 0
         
-        # Get text
-        if 'text' in article and article['text']: text = article['text']
-        else:
-            raise MissingDataException(f"Entry has no text.")
+    def contains_required_metadata(self, entry: Dict[str, Any]):
+        metadata_types = {
+            'id': str,
+            'source': str,
+            'title': str,
+            'url': str,
+            'date_published': str,
+            'authors': list, # unsure as of yet if this is a string or a list. TODO it is subject to change
+#           'summary': list  # see previous comment
```

Comment:
assume that it's a list of strings and that it will always be present (if not there's a bug in ARD)

Reply:
alright. Same for summary?

Reply:
yup. I just merged a PR that should have ensured that, and I updated the datasets on HF, so it should be ok. Let me know if I messed up :D